### ISSUE TITLE
Feature request: skipping descenders

### ISSUE THREAD

**Issue by skrat** [alacritty/alacritty#6785]
Per title, this would be super nice and would give alacritty a nice edge over competition.

A la https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration-skip-ink

**Comment by chrisduerr** [comment]
Why? How is this a "feature" that would benefit anyone?

**Comment by kchibisov** [comment]
>Why? How is this a "feature" that would benefit anyone?

It's pretty common and looks visually better? Because underline doesn't bleed into the glyph.

**Comment by skrat** [comment]
Without

![image](https://user-images.githubusercontent.com/5008/225436102-eccae8f6-e833-44a9-a1cb-6971a45f7608.png)

With

![image](https://user-images.githubusercontent.com/5008/225436225-6758432a-2d6f-4c52-925c-79d5399f541b.png)

If this is not obvious, why is the latter better, then you can't be helped.

**Comment by skrat** [comment]
Above screenshots are from the browser. I'm going to hitchike on this issue: how can I increase distance between underline and the glyph?

![image](https://user-images.githubusercontent.com/5008/225437065-9ee22b14-c79b-4f40-acad-e1fa355d5cee.png)

**Comment by kchibisov** [comment]
```
font:
  glyph_offset:
    y: -1
```

Or positive value for it, don't remember exactly.

**Comment by chrisduerr** [comment]
I don't see how this provides any positive value for Alacritty and it would certainly require drawing more. So it's just inefficient really.

@kchibisov are you actually interested in this?

**Comment by kchibisov** [comment]
It won't draw much more though and drawing is done in one call, so it doesn't really matter how many of them you'll have, you'll have data way less than you have for `grid` drawing maybe you'll loose like 1 frame per second at worse on a full screen of rectangles.

I'm sort of interested to at least try implementing it at some point, I do find it visually superior, since it'll help with letters like `щ` and `ш`, they look exactly the same, except for descender, and if you draw underline you won't see the difference between them at all with some fonts positioning underline too close.

**Comment by skrat** [comment]
It'd be worth trying to just skip adding underline rect for "descender glyphs". Would something like crossfont providing `descender: bool` for each `RasterizedGlyph` help?

**Comment by kchibisov** [comment]
Nah, I don't want a subpar solution for this. Because if you skip the entire underline under the cell it'll look off.

**Comment by skrat** [comment]
I'm no expert, but after some reading, most of the underline stuff happens in `rects.rs` and `mod.rs`. We would need to identify descender glyphs (crossfont change?) and cache a collection of start-end pairs to position the "rects"?

**Comment by kchibisov** [comment]
The most complex part will be inside the `crossfont`, code in alacritty will be sort of trivial, though, might be not in the current shape of alacritty's renderer... I'd suggest to focus on crossfont required changes, not on the stuff needed here in alacritty.

The hardest part is identifying the descender skipping is not really a problem. I'd suggest to look into present frameworks and APIs on how they do so, if you're interested. I know that skia can do so.

It would be also nice to know what `freetype`, `coretext`, and `directwrite` could provide us here.
